### PR TITLE
Add support to directly provide `crypto:PrivateKey` and `crypto:PublicKey` in JWT signature configurations

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes: [#6514](https://github.com/ballerina-platform/ballerina-library/issues/6514)

### Issuing self-signed JWT using `crypto:PrivateKey`
```ballerina
byte[] privateKeyContent = check io:fileReadBytes("path-to/private.key");
crypto:PrivateKey privateKey = check crypto:decodeRsaPrivateKeyFromContent(privateKeyContent);
jwt:IssuerConfig issuerConfig = {
    username: "John",
    issuer: "wso2",
    audience: ["ballerina", "ballerinaSamples"],
    expTime: 600,
    signatureConfig: {
        config: privateKey
    }
};
string jwt = check jwt:issue(issuerConfig);
```

### Validating self-signed JWT using `crypto:PublicKey`
```ballerina
byte[] pubicCertContent = check io:fileReadBytes("path-to/public.crt");
crypto:PublicKey publicKey = check crypto:decodeRsaPublicKeyFromContent(pubicCertContent);
jwt:ValidatorConfig validatorConfig = {
    issuer: "wso2",
    audience: ["ballerina", "ballerinaSamples"],
    clockSkew: 60,
    signatureConfig: {
        certFile: publicKey
    }
};
string jwtContent = "xxxxxx";
jwt:Payload result = check jwt:validate(jwtContent, validatorConfig);
```

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
- [x] Added tests
- [X] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

File: docs/proposals/enable-crypto-key-usage.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Proposal: Enable direct use of `crypto:PrivateKey` and `crypto:PublicKey` in JWT signature configurations
+
+_Authors_: @ayeshLK \
+_Reviewers_: @daneshk @NipunaRanasinghe @Bhashinee \
```
Comment (ayeshLK) [inline docs/proposals/enable-crypto-key-usage.md:4]:
```suggestion
_Reviewers_: @shafreenAnfar @daneshk @NipunaRanasinghe @Bhashinee \
```

File: ballerina/jwt_validator.bal [hunk 2]
```diff
@@ -57,7 +57,7 @@ public type ValidatorSignatureConfig record {|
         cache:CacheConfig cacheConfig?;
         ClientConfiguration clientConfig = {};
     |} jwksConfig?;
-    string certFile?;
+    string|crypto:PublicKey certFile?;
```
Comment (Bhashinee) [inline ballerina/jwt_validator.bal:60]: Shall we update the documentation as well? Saying that we can give a file path or a `crypto:PublicKey` record as the certificate?

Comment (ayeshLK) [reply ballerina/jwt_validator.bal:60]: Fixed with [857b997c9142b2a14d59a545823326dc49b86191](https://github.com/ballerina-platform/module-ballerina-jwt/pull/1229/commits/857b997c9142b2a14d59a545823326dc49b86191)

**Review by Bhashinee — APPROVED** [review]

File: ballerina/jwt_validator.bal [hunk 3]
```diff
@@ -48,7 +48,7 @@ public type ValidatorConfig record {
 # Represents JWT signature configurations.
 #
 # + jwksConfig - JWKS configurations
-# + certFile - Public certificate file
+# + certFile - Public certificate file or a `crypto:PublicKey`
```
Comment (Bhashinee) [inline ballerina/jwt_validator.bal:51]: Public certificate file path or a `crypto:PublicKey`

WDYT? @ayeshLK

Comment (ayeshLK) [reply ballerina/jwt_validator.bal]: Fixed with [010741eb9e2ea727d67d16db319fc93ce2d0843d](https://github.com/ballerina-platform/module-ballerina-jwt/pull/1229/commits/010741eb9e2ea727d67d16db319fc93ce2d0843d)

File: docs/proposals/enable-crypto-key-usage.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+_Updated_: 2024/05/08 \
+_Issue_: [#6515](https://github.com/ballerina-platform/ballerina-library/issues/6515)
+
+## Summary
+
+JWT signature configurations are designed to facilitate the generation and verification of JWT signatures. 
+Therefore, the JWT package should support direct usage of `crypto:PrivateKey` and `crypto:PublicKey` in 
+`jwt:IssuerSignatureConfig` and `jwt:ValidatorSignatureConfig` respectively.
+
+
```
Comment (NipunaRanasinghe) [inline docs/proposals/enable-crypto-key-usage.md:15]:
```suggestion

```

File: ballerina/tests/jwt_validator_test.bal [hunk 5]
```diff
@@ -17,6 +17,8 @@
 // NOTE: All the tokens/credentials used in this test are dummy tokens/credentials and used only for testing purposes.
 
 import ballerina/test;
+import ballerina/io;
+import ballerina/crypto;
```
Comment (NipunaRanasinghe) [inline ballerina/tests/jwt_validator_test.bal:21]: Imports are not ordered it seems. We have the imports formatting feature from update9.

Comment (NipunaRanasinghe) [reply ballerina/tests/jwt_validator_test.bal:21]: apply where necessary

**Review by NipunaRanasinghe — COMMENTED** [review]
LGTM

**Comment by NipunaRanasinghe** [comment]
@ayeshLK do we have a changelog in this module to be updated?

**Comment by ayeshLK** [comment]
> @ayeshLK do we have a changelog in this module to be updated?

Already updated

File: ballerina/tests/jwt_validator_test.bal (hunk 5 above)
Comment (ayeshLK) [reply ballerina/tests/jwt_validator_test.bal]: Fixed with [4171eb3c3fd1372742b8e0f62278ebd1ccdc3287](https://github.com/ballerina-platform/module-ballerina-jwt/pull/1229/commits/4171eb3c3fd1372742b8e0f62278ebd1ccdc3287)

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**